kroxylicious/src/main/java/io/kroxylicious/proxy/classloader/ClassloaderUtils.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.kroxylicious.proxy.classloader;
+
+import java.util.Optional;
+
+public class ClassloaderUtils {
+    public static <T> Optional<Class<? extends T>> loadClassIfPresent(Class<T> superClass, String binderClass) {
+        return loadClass(superClass, binderClass, false);
+    }
+
+    public static <T> Class<? extends T> loadRequiredClass(Class<T> superClass, String configurationHookClass) {
+        return loadClass(superClass, configurationHookClass, true)
+                .orElseThrow(() -> new IllegalArgumentException("could not load " + configurationHookClass));
+    }
+
+    public static <T> Optional<Class<? extends T>> loadClass(Class<T> superClass, String configurationHookClass, boolean failIfNotFound) {
+        try {
+            Class<?> targetClass = ClassloaderUtils.class.getClassLoader().loadClass(configurationHookClass);
+            if (superClass.isAssignableFrom(targetClass)) {
+                return Optional.of(targetClass.asSubclass(superClass));
+            }
+            else {
+                throw new RuntimeException("class " + targetClass.getSimpleName() + " is not assignable to " + superClass);
+            }
+        }
+        catch (ClassNotFoundException e) {
+            if (failIfNotFound) {
```

Comment:
Do we need this flag? Wouldn't callers who set it to true just do  
```
loadClass(superClass, configurationHookClass, true)
                 .orElseThrow(() -> new IllegalArgumentException("could not load " + configurationHookClass));
```

Reply:
refactored to only have a `loadClassIfPresent` method and leave it to callers to react to missing classes